### PR TITLE
If the default branch is not present do not report error on stats indexing

### LINKED ISSUE #15257: [E] stats queue indexer.Index(145) failed: object does not exist [id: refs/heads/master, rel_path: ]
- Gitea version (or commit ref): 1.13.6
- Git version: 2.24.4
- Operating system: CentOS 7

- Database (use `[x]`):
  - [X] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [X] No
- Log gist:
```
2021/04/02 17:22:33 ...eful/manager_unix.go:127:handleSignals() [W] PID 9116. Received SIGTERM. Shutting down...
2021/04/02 17:22:33 .../graceful/manager.go:217:doHammerTime() [W] Setting Hammer condition
2021/04/02 17:22:34 .../graceful/manager.go:231:doTerminate() [W] Terminating
2021/04/02 17:22:35 ...rm/session_schema.go:360:Sync2() [W] Table user Column max_repo_creation db default is '-1', struct default is -1
2021/04/02 17:22:36 ...dexer/stats/queue.go:24:handle() [E] stats queue indexer.Index(145) failed: object does not exist [id: refs/heads/master, rel_path: ]
```

## Description

I got the mentioned indexer error in the gitea log.

### PR DESCRIPTION
**Pull request by zeripath**
Just debug log instead

Fix #15257

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Review by silverwind — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

File: modules/indexer/stats/db.go [hunk 1]
```diff
@@ -38,8 +38,8 @@ func (db *DBIndexer) Index(id int64) error {
 	// Get latest commit for default branch
 	commitID, err := gitRepo.GetBranchCommitID(repo.DefaultBranch)
 	if err != nil {
-		log.Error("Unable to get commit ID for defaultbranch %s in %s", repo.DefaultBranch, repo.RepoPath())
-		return err
+		log.Debug("Unable to get commit ID for defaultbranch %s in %s ... skipping this repository", repo.DefaultBranch, repo.RepoPath())
```
Comment (lunny) [inline modules/indexer/stats/db.go]: We should do this according if the error type is `ErrBranchIsNotExist`.

Comment (zeripath) [reply modules/indexer/stats/db.go]: I'm not sure because really the error is not very helpful - but I've done as you requested.

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
Please send back port

**Comment by zeripath** [comment]
This doesn't actually work because the error is not a branch does not exist error - follow-up PR coming up.

**Pull request merged**